test/test_encryption.py

```diff
@@ -2202,6 +2202,85 @@ def test_05_roundtrip_encrypted_unindexed(self):
         self.assertEqual(decrypted, val)
 
 
+# https://github.com/mongodb/specifications/blob/072601/source/client-side-encryption/tests/README.rst#rewrap
+class TestRewrapWithSeparateClientEncryption(EncryptionIntegrationTest):
+
+    MASTER_KEYS = {
+        "aws": {
+            "region": "us-east-1",
+            "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+        },
+        "azure": {
+            "keyVaultEndpoint": "key-vault-csfle.vault.azure.net",
+            "keyName": "key-name-csfle",
+        },
+        "gcp": {
+            "projectId": "devprod-drivers",
+            "location": "global",
+            "keyRing": "key-ring-csfle",
+            "keyName": "key-name-csfle",
+        },
+        "kmip": {},
+    }
+
+    @client_context.require_no_standalone
+    @client_context.require_version_min(6, 0, -1)
+    def setUp(self):
+        super().setUp()
+
+    def test_rewrap(self):
+        for src_provider in ALL_KMS_PROVIDERS.keys():
+            for dst_provider in ALL_KMS_PROVIDERS.keys():
+                self.run_test(src_provider, dst_provider)
```

Comment:
Would this be a good case for using a subTest? https://docs.python.org/3/library/unittest.html#subtests

The benefit is better error reporting when a combination fails.

Reply:
Done